constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/Transition.java

```diff
@@ -269,27 +304,64 @@ void config(float position, float velocity, long start, float duration) {
                     mEngine = sl = new StopLogicEngine();
                 }
 
-                sl.config(position, 1, velocity,
+                sl.config(position, destination, velocity,
                         duration, mMaxAcceleration,
                         mMaxVelocity);
+                Utils.log("     destination  = " + destination);
+                Utils.log("            v in  = " + velocity);
+                Utils.log("            v out = " + sl.getVelocity());
+                lastp = position;
             } else {
                 SpringStopEngine sl;
                 if (mEngine instanceof SpringStopEngine) {
                     sl = (SpringStopEngine) mEngine;
                 } else {
                     mEngine = sl = new SpringStopEngine();
                 }
-                sl.springConfig(position, 1, velocity,
+
+                sl.springConfig(position, destination, velocity,
                         mSpringMass,
                         mSpringStiffness,
                         mSpringDamping,
                         mSpringStopThreshold, mSpringBoundary);
+
+
             }
         }
 
+        float lastp;
+
         public float getTouchUpProgress(long currentTime) {
```

Comment:
Add javadoc, document `currentTime` units. Or rename so that it's clear what the unit is, eg: `currentTimeNanos`